overcast/runner/__init__.py

```diff
@@ -128,6 +128,99 @@ def get_creds_from_env():
+        self.record_resource = lambda *args, **kwargs: None
+        self.name      = name
+        self.info      = info
+        self.runner    = runner
+        self.keypair   = keypair
+        self.userdata  = userdata
+        self.server_id = None
+        self.fip_ids   = set()
+        self.port_ids  = set()
+        self.fip_address = None
+        self.server_status = None
+        self.nc = self.runner.get_nova_client()
+        if self.info['image'] in self.runner.mappings.get('images', {}):
+            self.info['image'] = self.runner.mappings['images'][self.info['image']]
+
+        if self.info['flavor'] in self.runner.mappings.get('flavors', {}):
+            self.info['flavor'] = self.runner.mappings['flavors'][self.info['flavor']]
+        self.image = self.nc.images.get(self.info['image'])
+        self.flavor = self.nc.flavors.get(self.info['flavor'])
+
+    ##
+    # This one poll the server and return server status.
+    ##
+    def poll(self, desired_status = 'ACTIVE'):
+        if self.server_status != desired_status:
+            self.server_status = self.nc.servers.get(self.server_id).status
+        return self.server_status
+
+    ##
+    # Cleaner: This method remove server, fip, port etc.
+    #
+    # We could keep fip and may be ports (ports are getting deleted with current
+    # neutron client), but that is going to be bit more complex to make sure
+    # right port is assigned to right fip etc, so atm, just removing them.
+    ##
+    def clean(self):
+        for fip_id in self.fip_ids:
+            delete_floating_ip(fip_id)
+            self.fip_ids.discard(fip_id)
+
+        for port_id in self.port_ids:
+            delete_port(port_id)
+            self.port_ids.discard(port_id)
+
+        server = self.nc.servers.get(self.server_id)
+        server.delete()
+
+    def build(self):
+
+        def _map_network(network):
+            if network in self.runner.mappings.get('networks', {}):
+                netid = self.runner.mappings['networks'][network]
+            elif network in self.runner.networks:
+                netid = self.runner.networks[network]
+            else:
+                netid = network
+
+            return netid
+
+        nics = []
+        for eth_idx, network in enumerate(self.info['networks']):
+           port_name = '%s_eth%d' % (self.name, eth_idx)
+           port_id = self.runner.create_port(port_name, _map_network(network['network']),
+                                      [self.runner.secgroups[secgroup] for secgroup in network.get('secgroups', [])])
+           self.record_resource('port', port_id)
+
+           if network.get('assign_floating_ip', False):
+              fip_id, self.fip_address = self.runner.create_floating_ip()
+              self.runner.associate_floating_ip(port_id, fip_id)
+              self.fip_ids.add(fip_id)
+              self.port_ids.add(port_id)
+           else:
+               self.fip_address = None
```

Comment:
In the best case, this is superfluous. In the worst case, it overwrites information.

If there's only one interface and it doesn't want a floating ip, self.fip_address is already None (set in __init__.py).

If, however, there's more than one interface and the first one should get a floating ip and the second one shouldn't, you'll end up with self.fip_address == None.